### ISSUE TITLE
Kip-848: Next Generation Consumer Group

### ISSUE THREAD

**Issue by aratz-lasa** [twmb/franz-go#685]
I wanted to know what is the plan on implementing KIP-848. I see a dedicated branch just for this, but it's been inactive for a year. @twmb I would be happy to help move it forward.

**Comment by twmb** [comment]
I have to evaluate what this entails. The branch is stale because I _used to_ implement protocol support before it was rolled out in Kafka, but stopped once Kafka started changing APIs repeatedly before the API was finally released.

I'm not sure 848 is high priority though because it's released as preview only. I'm not sure how hard it'll be to implement, but likely pretty hard...

**Comment by twmb** [comment]
Yeah I'm probably going to need help with this, this is a big one.

**Comment by twmb** [comment]
I definitely missed your reply above^.

I'm starting to think about this (a _little_ bit). One thing I may lean to do first is to improve the guts of the consumer so that moving a partition between brokers does not require a full consumer stop/restart -- and then after, implement KIP-848.

Or it may make sense to just start with 848 logic, I'm really not sure -- I haven't dug into it much.

**Comment by twmb** [comment]
I'll do it alone it seems,

#912.

**Comment by twmb** [comment]
This will be in 1.19, closing now.
